lib/Doctrine/DBAL/Exception/DriverException.php

```diff
@@ -45,13 +45,7 @@ class DriverException extends DBALException
      */
     public function __construct($message, \Doctrine\DBAL\Driver\DriverException $driverException)
     {
-        $exception = null;
-
-        if ($driverException instanceof \Exception) {
```

Comment:
This makes perfect sense because we are type hinting against an interface in the constructor and need to make sure that we really pass an exception to the parent constructor (IIRC).